core/src/main/java/org/kiwix/kiwixmobile/core/main/OnSwipeTouchListener.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.GestureDetector
+import android.view.GestureDetector.SimpleOnGestureListener
+import android.view.MotionEvent
+import android.view.View
+import android.view.View.OnTouchListener
+import kotlin.math.abs
+
+open class OnSwipeTouchListener @SuppressLint("SyntheticAccessor")
+constructor(ctx: Context?) : OnTouchListener {
+  private val gestureDetector: GestureDetector
+
+  companion object {
+    private const val SWIPE_THRESHOLD = 100
+    private const val SWIPE_VELOCITY_THRESHOLD = 100
+  }
+
+  @SuppressLint("ClickableViewAccessibility") override fun onTouch(
```

Comment:
keep the annotations for methods on their own line and then we should be able to make these method signatures look a bit cleaner eg

```
@SuppressLint("ClickableViewAccessibility") 
override fun onTouch(v: View, event: MotionEvent): Boolean =
 gestureDetector.onTouchEvent(event)
```

Reply:
Sir, I will take care this thing in future